extra_data/components.py

```diff
@@ -262,6 +284,13 @@ class XtdfDetectorBase(MultimodDetectorBase):
     n_modules = 16
     _main_data_key = 'image.data'
 
+    def __init__(self, data: DataCollection, detector_name=None, modules=None,
+                 *, min_modules=1, n_modules=None):
+        if n_modules is not None:
+            raise NotImplementedError(
+                '--n-modules option is not available for AGIPD/DSSC/LPD data')
```

Comment:
I'm not super keen on having a parameter in the signature which can't actually be used. It works, but it feels a bit clumsy.